### DISCUSSION TITLE
Vacuum Entity: Add operating mode

### DISCUSSION THREAD

**Discussion by dougiteixeira** [home-assistant/frontend#11056] [Other feature requests]
Some robot vacuum cleaners have an control of operating mode, which usually creates a select entity for the device in the Home Assistant.

The suggestion is to create the mode property for the vacuum entity and include support for controlling the operating mode directly by the vacuum entity, including this property in the vaccum card.

![image](https://user-images.githubusercontent.com/31328123/147798532-9ea04857-6093-4f42-8317-c77f069a3c1f.png)

**Comment by frenck** [comment]
Which integrations could benefit from this?

**Reply by dougiteixeira** [reply]
@frenck,

Just by analyzing the code it is difficult to determine the extension of the benefited integrations, but I could verify analysing of the code of the integrations, what has something related to the vacuum mode of operation are: Tuya and Shark IQ.

[Tuya ](https://www.home-assistant.io/integrations/tuya/)([select.py](https://github.com/home-assistant/core/blob/06b9be4d9cd35ad07a8f9ab6c4c1e71100d33754/homeassistant/components/tuya/select.py#L242))
![image](https://user-images.githubusercontent.com/31328123/147866846-faee6d97-e696-4eab-9bea-0cb433875ee8.png)

[Shark IQ](https://www.home-assistant.io/integrations/sharkiq/) ([vacuum.py](https://github.com/home-assistant/core/blob/06b9be4d9cd35ad07a8f9ab6c4c1e71100d33754/homeassistant/components/sharkiq/vacuum.py#L154))
![image](https://user-images.githubusercontent.com/31328123/147866535-c6830c50-d1bb-41ab-803e-61f2f2defa85.png)

But from previous contact I've had with vacuum robots, most have some sort of control over the cleaning mode, whether it's to determine whether it's a quick, edge-focused, or automatic cleaning.

And the benefit has to be thought about usability for the user, although this can be implemented using other types of entities, the user would not have full control of the vacuum by the native card. This addition would prevent the user from having to use other screen elements to configure the desired operating mode.

And as it will be a feature that will have compatibility control (supported features) it would have no impact on integrations that do not have such feature today, but which, with the inclusion in the documentation, may come to be implemented in the future.

Thanks for taking the time to consider this.

**Reply by frenck** [reply]
The Tuya operating mode is actually a bad example, as it conflicts with the vacuum entity state already (hence is should not be part of the vacuum entity).

Could you describe the what the operation mode entails?


Please note that two integrations (assuming Tuya will work) is probably not enough to warrant such an addition.

**Comment by dougiteixeira** [comment]
I did some tests implementing this in core and it works fine (I didn't implement it in fronted).

[Here is the commit link](https://github.com/dougiteixeira/core/commit/7fd5717f019448b2e7021c566d13b34b75baff79) if you want to check the implications of adding such a resource.
